src/helics/shared_api_library/ValueFederate.h

```diff
@@ -21,37 +21,36 @@ extern "C" {
 #endif
 
 /**
- * sub/pub registration
+ * input/publication registration
  */
 
 /**
- * Create a subscription.
+ * Create an input and add a publication target.
  *
- * @details The subscription becomes part of the federate and is destroyed when the federate is freed so there are no separate free
- * functions for subscriptions and publications.
+ * @details this method is a wrapper method to create and unnamed input and add a publication target to it
  *
- * @param fed The federate object in which to create a subscription, must have been created with /ref helicsCreateValueFederate or
+ * @param fed The federate object in which to create an input, must have been created with /ref helicsCreateValueFederate or
  * /ref helicsCreateCombinationFederate.
- * @param key The identifier matching a publication to get a subscription for.
- * @param units A string listing the units of the subscription (may be NULL).
+ * @param key The identifier matching a publication to add as an input target.
+ * @param units A string listing the units of the input (may be NULL).
  *
  * @param[in,out] err A pointer to an error object for catching errors.
 
  *
- * @return An object containing the subscription.
+ * @return An object containing the input.
  */
 HELICS_EXPORT HelicsInput helicsFederateRegisterSubscription(HelicsFederate fed, const char* key, const char* units, HelicsError* err);
 
 /**
  * Register a publication with a known type.
  *
  * @details The publication becomes part of the federate and is destroyed when the federate is freed so there are no separate free
- * functions for subscriptions and publications.
+ * functions for inputs and publications.
  *
  * @param fed The federate object in which to create a publication.
  * @param key The identifier for the publication the global publication key will be prepended with the federate name.
  * @param type A code identifying the type of the input see /ref HelicsDataTypes for available options.
- * @param units A string listing the units of the subscription (may be NULL).
+ * @param units A string listing the units of the publication (may be NULL).
```

Comment:
I don't know this API well but shouldn't this be "input"?

Reply:
I will double check, but there was subscription listed in may place it shouldn't have been, this one is helicsFederateRegisterPublication so publication is correct